### PR TITLE
Add adr about rke2-flannel

### PR DESCRIPTION
**Pull request by manuelbuil**
#### Proposed Changes ####

ADR. I am proposing a new feature

#### Types of Changes ####

New feature proposal

#### Linked Issues ####

https://github.com/rancher/rke2/issues/5215

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: docs/adrs/006-add-flannel-support-in-windows.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+### Strength
+
+* We can offer a plan B for customers that can't afford getting TCP Resets
+* We have an alternative while Microsoft and Tigera fix the problem
+* We are maintainers of flannel and could support it easily
+
+### Weakness
+* Yet another cni plugin to support
+* Flannel is very limited. We should document it very well to avoid disappointment on non-knowledgeable customers
+* How to make it only available for rke2-Windows?
```
Comment (brandond) [inline docs/adrs/006-add-flannel-support-in-windows.md:35]: Is it even possible to run different CNIs on different nodes? If we support flannel for Windows, my understanding is that we've need to support it for the Linux nodes as well.

File: docs/adrs/006-add-flannel-support-in-windows.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+### Strength
+
+* We can offer a plan B for customers that can't afford getting TCP Resets
+* We have an alternative while Microsoft and Tigera fix the problem
+* We are maintainers of flannel and could support it easily
+
+### Weakness
+* Yet another cni plugin to support
+* Flannel is very limited. We should document it very well to avoid disappointment on non-knowledgeable customers
```
Comment (brandond) [inline docs/adrs/006-add-flannel-support-in-windows.md:34]: Would we support a network policy controller in this configuration, or would clusters configured this way lack netpol support?

File: docs/adrs/006-add-flannel-support-in-windows.md (hunk 1 above)
Comment (rbrtbnfgl) [reply docs/adrs/006-add-flannel-support-in-windows.md:35]: Yes this is the idea to support flannel as CNI of RKE2 for both linux and windows.

File: docs/adrs/006-add-flannel-support-in-windows.md (hunk 2 above)
Comment (rbrtbnfgl) [reply docs/adrs/006-add-flannel-support-in-windows.md:34]: It doesn't support any network policy. The only CNI with network policy and windows support is calico that has the TCP reset issue.

File: docs/adrs/006-add-flannel-support-in-windows.md (hunk 1 above)
Comment (brandond) [reply docs/adrs/006-add-flannel-support-in-windows.md:35]: Should we remove this as a bullet point then?

Comment (manuelbuil) [reply docs/adrs/006-add-flannel-support-in-windows.md:35]: yeah true, I was thinking on not supporting it on "all-linux" nodes. But it does not make sense if we anyway need to support it in linux. I'll remove it

File: docs/adrs/006-add-flannel-support-in-windows.md (hunk 2 above)
Comment (manuelbuil) [reply docs/adrs/006-add-flannel-support-in-windows.md:34]: No network policies. Customers know about this and still they prefer that compared to Calico

**Review by dereknola — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Review by matttrach — COMMENTED** [review]
While I agree that this makes a lot of sense, I would like to hear what @briandowns and @rancher-max has to say about it since it would expand our support matrix and possibly change our security assumptions.

**Review by rancher-max — APPROVED** [review]
I am fine with this. I do agree with you point about documentation -- it should be extremely thorough. I also wonder if we have to use an "rke2-flannel" directly or can we double dip from k3s and pull that in? Or is that what this would effectively be, just packaged as a helm chart instead of embedded?

**Comment by brandond** [comment]
Ideally it would be a helm chart, same as the other CNIs.

**Comment by manuelbuil** [comment]
> I am fine with this. I do agree with you point about documentation -- it should be extremely thorough. I also wonder if we have to use an "rke2-flannel" directly or can we double dip from k3s and pull that in? Or is that what this would effectively be, just packaged as a helm chart instead of embedded?

Helm chart (in linux)! In Windows, I still need to figure it out but probably it will follow Calico's way... AFAIK, Windows does not support pods with access to the root network namespace, which is a requirement for CNI plugin pods. If that is still the case, we can't use helm chart in Windows and, as in Calico, we will need to execute flannel binary (flanneld.exe) from within rke2.exe

**Comment by manuelbuil** [comment]
> While I agree that this makes a lot of sense, I would like to hear what @briandowns and @rancher-max has to say about it since it would expand our support matrix and possibly change our security assumptions.

Thanks Matt. What security assumptions? I haven't thought about those

**Comment by brandond** [comment]
> we can't use helm chart in Windows and, as in Calico, we will need to execute flannel binary (flanneld.exe) from within rke2.exe

Yeah, as long as we set it up like Calico, where the Linux nodes use the chart, and the Windows nodes key off the charts presence to run the binary, I think we should be fine.

**Comment by manuelbuil** [comment]
> > we can't use helm chart in Windows and, as in Calico, we will need to execute flannel binary (flanneld.exe) from within rke2.exe
> 
> Yeah, as long as we set it up like Calico, where the Linux nodes use the chart, and the Windows nodes key off the charts presence to run the binary, I think we should be fine.

that's the idea

**Comment by matttrach** [comment]
I was thinking there is probably some security assumptions in the use of network policies. Without network policies won't pods be basically open for connections?

**Comment by manuelbuil** [comment]
> I was thinking there is probably some security assumptions in the use of network policies. Without network policies won't pods be basically open for connections?

Inside the cluster, there will be no limitation for connections to the pods. Connectivity from the outside to the pod is still limited. In any case, we will document that network policies are not available in flannel, users will be aware of this

**Comment by manuelbuil** [comment]
As nobody seems against it, I'll create an Issue and cover all required PRs there

**Review by matttrach — APPROVED** [review]

**Pull request merged**